flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/context/SessionContextTest.java

```diff
@@ -126,61 +140,87 @@ public void testSetAndResetKeyInConfigOptions() throws Exception {
-        assertEquals("default", getConfiguration(sessionContext).getString(TABLE_SQL_DIALECT));
+        assertEquals("default", getConfiguration().getString(TABLE_SQL_DIALECT));
 
         sessionContext.reset(MAX_PARALLELISM.key());
-        assertEquals(16, getConfiguration(sessionContext).getInteger(MAX_PARALLELISM));
+        assertEquals(16, getConfiguration().getInteger(MAX_PARALLELISM));
 
         sessionContext.reset(NAME.key());
-        assertNull(getConfiguration(sessionContext).get(NAME));
+        assertNull(getConfiguration().get(NAME));
 
         sessionContext.reset(OBJECT_REUSE.key());
-        assertTrue(getConfiguration(sessionContext).getBoolean(OBJECT_REUSE));
+        assertTrue(getConfiguration().getBoolean(OBJECT_REUSE));
     }
 
     @Test
-    public void testSetWithConfigOptionAndResetWithYamlKey() throws Exception {
-        SessionContext sessionContext = createSessionContext();
+    public void testSetWithConfigOptionAndResetWithYamlKey() {
         // runtime config option and has deprecated key
         sessionContext.set(TABLE_PLANNER.key(), "blink");
-        assertEquals(
-                "blink", getConfiguration(sessionContext).get(TABLE_PLANNER).name().toLowerCase());
+        assertEquals("blink", getConfiguration().get(TABLE_PLANNER).name().toLowerCase());
 
         sessionContext.reset(TABLE_PLANNER.key());
-        assertEquals(
-                "old", getConfiguration(sessionContext).get(TABLE_PLANNER).name().toLowerCase());
-        assertEquals(
-                "old", getConfigurationMap(sessionContext).get("execution.planner").toLowerCase());
+        assertEquals("old", getConfiguration().get(TABLE_PLANNER).name().toLowerCase());
+        assertEquals("old", getConfigurationMap().get("execution.planner").toLowerCase());
     }
 
     @Test
-    public void testSetAndResetKeyNotInYaml() throws Exception {
-        SessionContext sessionContext = createSessionContext();
+    public void testSetAndResetKeyNotInYaml() {
         // other property not in yaml and flink-conf
         sessionContext.set("aa", "11");
         sessionContext.set("bb", "22");
 
-        assertEquals("11", getConfigurationMap(sessionContext).get("aa"));
-        assertEquals("22", getConfigurationMap(sessionContext).get("bb"));
+        assertEquals("11", getConfigurationMap().get("aa"));
+        assertEquals("22", getConfigurationMap().get("bb"));
 
         sessionContext.reset("aa");
-        assertNull(getConfigurationMap(sessionContext).get("aa"));
-        assertEquals("22", getConfigurationMap(sessionContext).get("bb"));
+        assertNull(getConfigurationMap().get("aa"));
+        assertEquals("22", getConfigurationMap().get("bb"));
 
         sessionContext.reset("bb");
-        assertNull(getConfigurationMap(sessionContext).get("bb"));
+        assertNull(getConfigurationMap().get("bb"));
+    }
+
+    @Test
+    public void testAddJar() throws IOException {
+        sessionContext.addJar(udfJar.getPath());
```

Comment:
Please also add tests which the jar path is prefixed with `file://` and relative path `./xxx`.

You can use `new File(".").getAbsoluteFile().toPath().relativize(udfJar.toPath())` to get a relative path to the current working dir.